xline/src/storage/lease_store/mod.rs

```diff
@@ -0,0 +1,537 @@
+    /// Current node is leader or not
+    state: Arc<State>,
+    /// Header generator
+    header_gen: Arc<HeaderGenerator>,
+}
+
+impl LeaseStore {
+    /// New `KvStore`
+    #[allow(clippy::integer_arithmetic)] // Introduced by tokio::select!
+    pub(crate) fn new(
+        del_tx: mpsc::Sender<DeleteMessage>,
+        mut lease_cmd_rx: mpsc::Receiver<LeaseMessage>,
+        state: Arc<State>,
+        header_gen: Arc<HeaderGenerator>,
+    ) -> Self {
+        let inner = Arc::new(LeaseStoreBackend::new(del_tx, state, header_gen));
+        let _handle = tokio::spawn({
+            let inner = Arc::clone(&inner);
+            async move {
+                while let Some(lease_msg) = lease_cmd_rx.recv().await {
+                    match lease_msg {
+                        LeaseMessage::Attach(tx, lease_id, key) => {
+                            assert!(
+                                tx.send(inner.attach(lease_id, key)).is_ok(),
+                                "receiver is closed"
+                            );
+                        }
+                        LeaseMessage::Detach(tx, lease_id, key) => {
+                            assert!(
+                                tx.send(inner.detach(lease_id, &key)).is_ok(),
+                                "receiver is closed"
+                            );
+                        }
+                        LeaseMessage::GetLease(tx, key) => {
+                            assert!(tx.send(inner.get_lease(&key)).is_ok(), "receiver is closed");
+                        }
+                        LeaseMessage::LookUp(tx, lease_id) => {
+                            assert!(
+                                tx.send(inner.look_up(lease_id)).is_ok(),
+                                "receiver is closed"
+                            );
+                        }
+                    }
+                }
+            }
+        });
+        Self { inner }
+    }
+
+    /// execute a auth request
+    pub(crate) fn execute(
+        &self,
+        id: ProposeId,
+        request: RequestWithToken,
+    ) -> Result<CommandResponse, ExecuteError> {
+        self.inner
+            .handle_lease_requests(id, request.request)
+            .map(CommandResponse::new)
+    }
+
+    /// sync a auth request
+    pub(crate) async fn after_sync(&self, id: &ProposeId) -> SyncResponse {
+        SyncResponse::new(self.inner.sync_request(id).await)
+    }
+
+    /// Check if the node is leader
+    fn is_leader(&self) -> bool {
+        self.inner.is_leader()
+    }
+
+    /// Get lease by id
+    pub(crate) fn look_up(&self, lease_id: i64) -> Option<Lease> {
+        self.inner.look_up(lease_id)
+    }
+
+    /// Get all leases
+    pub(crate) fn leases(&self) -> Vec<Lease> {
+        let mut leases = self
+            .inner
+            .lease_collection
+            .read()
+            .lease_map
+            .values()
+            .cloned()
+            .collect::<Vec<_>>();
```

Comment:
`collect_vec`

Reply:
I think this is unnecessary, and in other places in our project, we also use `.collect::<Vec<_>>()`